CMake link directories issue

Building with CMake fails with 1.48 on OpenBSD.

    FAILED: nestopia 
    : && /usr/ports/pobj/nestopia-1.48/bin/c++  -O2 -pipe -g -g   [filenames]  -o nestopia -L/usr/local/lib -lz -larchive -lao -lSDL2 -lepoxy -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl -Wl,-rpath-link,/usr/X11R6/lib:/usr/local/lib && :
    /usr/bin/ld: cannot find -lepoxy
    c++: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.

The trouble is that libepoxy is located in `/usr/X11R6/lib` on OpenBSD, but that directory is not passed via `-L`.

pkg-config does the right thing, so I’m not sure why CMake doesn’t pick it up:

    $ pkg-config --libs epoxy
    -L/usr/X11R6/lib -lepoxy

CMake configure output:

    -- The C compiler identification is Clang 4.0.0
    -- The CXX compiler identification is Clang 4.0.0
    -- Check for working C compiler: /usr/ports/pobj/nestopia-1.48/bin/cc
    -- Check for working C compiler: /usr/ports/pobj/nestopia-1.48/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/ports/pobj/nestopia-1.48/bin/c++
    -- Check for working CXX compiler: /usr/ports/pobj/nestopia-1.48/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Performing Test HAS_NO_NARROWING
    -- Performing Test HAS_NO_NARROWING - Success
    -- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1") 
    -- Found ZLIB: /usr/lib/libz.so.5.0 (found version "1.2.3") 
    -- Found LibArchive: /usr/local/lib/libarchive.so.10.1 (found version "3.3.2") 
    -- Checking for module 'ao'
    --   Found ao, version 1.2.0
    -- Checking for module 'sdl2'
    --   Found sdl2, version 2.0.5
    -- Checking for module 'epoxy'
    --   Found epoxy, version 1.3.1
    -- Checking for module 'gtk+-3.0'
    --   Found gtk+-3.0, version 3.22.21
    -- Performing Test HAS_NO_DEPRECATED_DECLARATIONS
    -- Performing Test HAS_NO_DEPRECATED_DECLARATIONS - Success
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /usr/ports/pobj/nestopia-1.48/build-amd64

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -353,18 +353,18 @@ target_link_libraries( nestopia ${LibArchive_LIBRARIES} )
 
 # libao
 pkg_check_modules( LibAO REQUIRED ao )
-target_include_directories( nestopia PRIVATE ${LibAO_INCLUDE_DIRS} )
-target_link_libraries( nestopia ${LibAO_LIBRARIES} )
+target_compile_options( nestopia PRIVATE ${LibAO_CFLAGS} )
+target_link_libraries( nestopia ${LibAO_LDFLAGS} )
 
 # SDL2
 pkg_check_modules( SDL2 REQUIRED sdl2 )
-target_include_directories( nestopia PRIVATE ${SDL2_INCLUDE_DIRS} )
-target_link_libraries( nestopia ${SDL2_LIBRARIES} )
+target_compile_options( nestopia PRIVATE ${SDL2_CFLAGS} )
+target_link_libraries( nestopia ${SDL2_LDFLAGS} )
 
 # LibEpoxy
 pkg_check_modules( LibEpoxy REQUIRED epoxy )
-target_include_directories( nestopia PRIVATE ${LibEpoxy_INCLUDE_DIRS} )
-target_link_libraries( nestopia ${LibEpoxy_LIBRARIES} )
+target_compile_options( nestopia PRIVATE ${LibEpoxy_CFLAGS} )
+target_link_libraries( nestopia ${LibEpoxy_LDFLAGS} )
 
 if( MINGW )
 	# Building for MinGW
@@ -384,8 +384,8 @@ elseif( UNIX )
 		target_compile_definitions( nestopia PRIVATE -D_GTK )
 
 		pkg_check_modules( GTK3 REQUIRED gtk+-3.0 )
-		target_include_directories( nestopia PRIVATE ${GTK3_INCLUDE_DIRS} )
-		target_link_libraries( nestopia ${GTK3_LIBRARIES} )
+		target_compile_options( nestopia PRIVATE ${GTK3_CFLAGS} )
+		target_link_libraries( nestopia ${GTK3_LDFLAGS} )
 
 		# check whether compiler accepts -Wno-deprecated-declarations
 		# in order to silence GTK+3 deprecation warnings

